src/main/java/com/google/devtools/build/lib/authandtls/NetrcParser.java

```diff
@@ -0,0 +1,245 @@
+// Copyright 2020 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.devtools.build.lib.authandtls;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.devtools.build.lib.authandtls.Netrc.Credential;
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/** A parser used to parse .netrc content */
```

Comment:
Is there any documentation on the format? If so, please add a link to it.

Reply:
Here's an example of netrc file documentation - https://man.cx/netrc(4)

A useful source in terms of documentation might be https://docs.python.org/3.8/library/netrc.html

Reply:
Added.

Reply:
Maybe link to the existing Starlark netrc parser in this repo: https://github.com/bazelbuild/bazel/blob/4c0febc02fab5e9fda3f40532a62854fb1c45631/tools/build_defs/repo/utils.bzl#L203-L204

Reply:
Added.